m3u8/reader_test.go

```diff
@@ -1037,6 +1037,34 @@ func TestDecodeMediaPlaylistWithDefines(t *testing.T) {
 	is.Equal(p.Defines[2].Value, "")
 }
 
+func TestDecodeMediaPlaylistWithGaps(t *testing.T) {
+	data := []struct {
+		playlist string
+		wantErr  bool
+	}{
+		{"sample-playlists/media-playlist-with-gap.m3u8", false},
+		{"sample-playlists/bad-media-playlist-with-gap.m3u8", true},
+	}
+	for _, d := range data {
+		t.Run(d.playlist, func(t *testing.T) {
+			is := is.New(t)
+			f, err := os.Open(d.playlist)
+			is.NoErr(err) // must open file
+			p, err := NewMediaPlaylist(0, 1)
+			is.NoErr(err) // must create playlist
+			err = p.DecodeFrom(bufio.NewReader(f), true)
+			if d.wantErr {
+				is.True(err != nil) // must return an error
+			} else {
+				is.NoErr(err) // must decode playlist
+				//Check parsed values
+				is.True(p.Segments[0].Gap)
```

Comment:
The is package uses comments on the same line in the error message. I therefore suggest to remove the // "Check parsed values" and instead add comments like "// the first segment should signal GAP" etc